src/wallet/wallet.h

```diff
@@ -1069,6 +1069,9 @@ class CWallet : public CCryptoKeyStore, public CValidationInterface
     boost::signals2::signal<void (CWallet *wallet, const uint256 &hashTx,
             ChangeType status)> NotifyTransactionChanged;
 
+    boost::signals2::signal<void (const CTransactionRef &ptxn,
```

Comment:
I think you should make this a static variable to make the zmq code simpler and less fragile and remove the dependency on ::vpwallets. Advantages of making this static:

* It will let you get rid of the `ConnectToWalletSignals` method and all the ifdefed code around it and simply connect to the signal in `CZMQNotificationInterface::Initialize` consistent with you how currently disconnect in `CZMQNotificationInterface::Shutdown`
* If will make it possible in the future for new wallets to be added to ::vpwallets at runtime without having to modify  zmq code and make it register for new notifications.